src/psyclone/dynamo0p1.py

```diff
@@ -233,11 +233,28 @@ def __init__(self, parent=None, loop_type=""):
 
         # Work out the variable name from  the loop type
         if self._loop_type == "colours":
-            self._variable_name = "colour"
+            tag = "colours_loop_idx"
+            suggested_name = "colour"
         elif self._loop_type == "colour":
-            self._variable_name = "cell"
+            tag = "colour_loop_idx"
+            suggested_name = "cell"
         else:
-            self._variable_name = "cell"
+            tag = "cell_loop_idx"
+            suggested_name = "cell"
+
+        # This will return the symbol table from the closest ancestor
+        # that contains one. However, the original symbol my be in a
+        # different symbol table and if this is the case we will end
+        # up declaring the variable twice. Issue #630 describes this
+        # problem.
+        symtab = self.find_symbol_table()
+        try:
+            data_symbol = symtab.lookup_with_tag(tag)
+        except KeyError:
+            name = symtab.new_symbol_name(suggested_name)
+            data_symbol = DataSymbol(name, INTEGER_TYPE)
+            symtab.add(data_symbol, tag=tag)
+        self.variable = Reference(data_symbol)
```

Comment:
Remove Reference()

Reply:
Oops. The reason I missed these is that they are not picked up by the tests. The reason they are not picked up by the tests is that the `gen()` method is used to generate code not the Fortran back-end and this uses `f2pygen`'s `DoGen` which takes a string and does this by using `object.name`. It so happens that `Reference.name` returns Reference._symbol.name so it does not matter whether we provide a Reference to the Symbol or the Symbol when using the `gen()` method. Anyway, I've modified the code :-).